python/ray/tune/tests/test_ray_trial_executor.py

```diff
@@ -43,6 +50,28 @@ def testSaveRestore(self):
         self.trial_executor.stop_trial(trial)
         self.assertEqual(Trial.TERMINATED, trial.status)
 
+    def testSaveRestoreTimeout(self):
+        trial = Trial("__fake")
+        self.trial_executor.start_trial(trial)
+        self.assertEqual(Trial.RUNNING, trial.status)
+        self.trial_executor.save(trial, Checkpoint.DISK)
+        self.trial_executor.set_status(trial, Trial.PAUSED)
+
+        ray_get = ray.get
+        start_trial = self.trial_executor._start_trial
+
+        # Timeout on first two attempts, then succeed on subsequent gets.
+        side_effects = [RayTimeoutError, RayTimeoutError, ray_get, ray_get]
+        with patch("ray.get", side_effect=side_effects), \
+             patch.object(self.trial_executor, "_start_trial") as mock_start:
+            mock_start.side_effect = start_trial
```

Comment:
why do you need to patch `_start_trial` if it is going to have the same side effect?

Reply:
this just gets us the call count (see below)